instrumentation/opentelemetry-instrumentation-cherrypy/src/opentelemetry/instrumentation/cherrypy/__init__.py

```diff
@@ -0,0 +1,140 @@
+from email.policy import default
+from logging import getLogger
+from time import time_ns
+from timeit import default_timer
+from typing import Collection
+
+from opentelemetry.util.http import parse_excluded_urls, get_excluded_urls
+import cherrypy
+from opentelemetry.instrumentation.cherrypy.package import _instruments
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.utils import _start_internal_or_server_span
+import opentelemetry.instrumentation.wsgi as otel_wsgi
+from opentelemetry.instrumentation.propagators import (
+    get_global_response_propagator,
+)
+from opentelemetry import trace, context
+from opentelemetry.semconv.trace import SpanAttributes
+from opentelemetry.instrumentation.falcon.version import __version__
+from opentelemetry.metrics import get_meter
+
+
+_logger = getLogger(__name__)
+_excluded_urls_from_env = get_excluded_urls("CHERRYPY")
+
+class CherryPyInstrumentor(BaseInstrumentor):
+    """An instrumentor for FastAPI
+
+    See `BaseInstrumentor`
+    """
+
+    def instrumentation_dependencies(self) -> Collection[str]:
+        return _instruments
+
+    def _instrument(self, **kwargs):
+        self._original_cherrypy_application = cherrypy._cptree.Application
+        cherrypy._cptree.Application = _InstrumentedCherryPyApplication
+        cherrypy.Application = _InstrumentedCherryPyApplication
```

Comment:
https://docs.cherrypy.dev/en/latest/extend.html#per-request-functions

Reply:
Issue with these hooks are that they needed to be called for every paths that we want to use them for. They do not apply directly to everywhere. [source](https://docs.cherrypy.dev/en/latest/extend.html#per-request-functions:~:text=Using%20that%20tool%20is%20as%20simple%20as%20follows%3A)